### ISSUE TITLE
stability: split constantly failing with "range is already split"

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#5501]
Sha: 66fc410

Nodes are constantly trying to split a range, but failing with "range is already split":

This happened on two different nodes:
Node1 starts showing those errors:
```
I160323 13:37:53.162930 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67112680 max=67108864
E160323 13:37:53.299001 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
I160323 13:37:53.304540 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67113121 max=67108864
E160323 13:37:53.433139 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
```

The range then moves to node0 which shows the same errors:
```
I160323 13:40:54.199611 storage/replica_command.go:1306  range 54: new leader lease replica {1 1 1} 2016-03-23 13:40:53.950131789 +0000 UTC 1.238253678s
I160323 13:40:54.204929 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67285543 max=67108864
E160323 13:40:54.940910 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
I160323 13:40:54.944791 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67285543 max=67108864
E160323 13:40:55.015126 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
I160323 13:40:55.647813 storage/gc_queue.go:341  [gc] completed with stats {Now:1458740454.203056445,0 Policy:{TTLSeconds:86400} GCKeys:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCPending:0 SequenceSpanTotal:0 SequenceSpanConsidered:0 SequenceSpanGCNum:0 PushTxn:0 ResolveTotal:0 ResolveSuccess:0}
I160323 13:40:55.653281 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67285543 max=67108864
E160323 13:40:55.702820 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
I160323 13:40:55.704203 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67285543 max=67108864 
E160323 13:40:55.826843 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
```

And finally back on node1:
```
I160323 13:40:59.797607 storage/replica_command.go:1306  range 54: new leader lease replica {2 2 2} 2016-03-23 13:40:56.648446961 +0000 UTC 4.123275865s
I160323 13:40:59.800715 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67285543 max=67108864
E160323 13:40:59.913447 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
I160323 13:40:59.914338 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67285984 max=67108864
E160323 13:41:00.048295 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1597: range is already split at key /System/StatusNode/4
I160323 13:41:00.571979 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67286425 max=67108864
```

Full logs for nodes 0 and 1:
[node0.log.parse.txt](https://github.com/cockroachdb/cockroach/files/186153/node0.log.parse.txt)
[node1.log.parse.txt](https://github.com/cockroachdb/cockroach/files/186154/node1.log.parse.txt)

All data and logs have been backed up on each node to: `/mnt/data.5501` `~/logs.5501`

**Comment by tschottdorf** [comment]
nodes 1 and 3 also complain about node2 periodically:

```
W160323 16:51:57.371958 storage/replica.go:1307  group 1 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                             │1.1 MiB, keepcost=18 KiB
W160323 16:51:57.372015 storage/replica.go:1307  group 84 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:14.548428 storage/replica_command.go:1306  range 1: new leader lease replica {2 2 2} 2016-03-23 16:51:12.157507597 +0000 UTC 3.38894334s
W160323 16:51:57.372063 storage/replica.go:1307  group 82 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:20.198526 server/status/runtime.go:144  runtime stats: 349 goroutines, 8.19MiB active, 1192.32cgo/sec, 0.06/0.00 %(u/s)time, 0.00 %gc (5x)
W160323 16:51:57.372110 storage/replica.go:1307  group 50 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:20.198700 server/status/linuxstats.go:56  mallinfo stats: ordblks=1.2 KiB, smblks=269 B, hblks=0 B, hblkhd=0 B, usmblks=0 B, fsmblks=19 KiB, uordblks=1.0 GiB, fordblks=
W160323 16:51:57.372257 storage/replica.go:1307  group 1 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                             │1.1 MiB, keepcost=19 KiB
W160323 16:51:57.372421 storage/replica.go:1307  group 14 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:20.322175 storage/replica_command.go:1306  range 18: new leader lease replica {2 2 2} 2016-03-23 16:51:17.513882587 +0000 UTC 3.806796409s
W160323 16:51:57.372568 storage/replica.go:1307  group 55 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:24.417306 storage/replica_command.go:1306  range 1: new leader lease replica {2 2 2} 2016-03-23 16:51:22.559364431 +0000 UTC 2.856224155s
W160323 16:51:57.372646 storage/replica.go:1307  group 29 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:30.198738 server/status/runtime.go:144  runtime stats: 350 goroutines, 10.02MiB active, 1318.87cgo/sec, 0.07/0.01 %(u/s)time, 0.00 %gc (6x)
W160323 16:51:57.372722 storage/replica.go:1307  group 4 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                             │I160323 16:51:30.198927 server/status/linuxstats.go:56  mallinfo stats: ordblks=1.2 KiB, smblks=250 B, hblks=0 B, hblkhd=0 B, usmblks=0 B, fsmblks=18 KiB, uordblks=1.0 GiB, fordblks=
W160323 16:51:57.372772 storage/replica.go:1307  group 8 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                             │1.1 MiB, keepcost=18 KiB
W160323 16:51:57.372846 storage/replica.go:1307  group 26 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                            │I160323 16:51:30.319914 storage/replica_command.go:1306  range 18: new leader lease replica {2 2 2} 2016-03-23 16:51:25.738843012 +0000 UTC 5.579484708s
W160323 16:51:57.372926 storage/replica.go:1307  group 102 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full                           │I160323 16:51:34.041569 storage/replica_command.go:1306  range 1: new leader lease replica {2 2 2} 2016-03-23 16:51:32.304555144 +0000 UTC 2.735421973s
W160323 16:51:57.372975 storage/replica.go:1307  group 6 on store 1 failed to send message to 2: storage/raft_transport.go:280: queue for node 2 is full
```

nodes 1 and 3 have in the 100s of cgo ops per second (the repeated split key search should certainly cause this, but why on two nodes?). node 2 and 4 have very few.

**Comment by tschottdorf** [comment]
two nodes have `Node.Batch` tasks piling up.

```
ec2-54-209-133-121.compute-1.amazonaws.com:
0xc8202c7a40: 8 tasks
4      server/node.go:686
2      storage/queue.go:381
1      storage/intent_resolver.go:275
1      kv/txn_coord_sender.go:823
ec2-54-209-69-52.compute-1.amazonaws.com:
0xc8202f1ce0: 339 tasks
335    server/node.go:686
2      storage/queue.go:381
1      ts/db.go:100
1      storage/intent_resolver.go:275
ec2-54-208-199-194.compute-1.amazonaws.com:
0xc82031c380: 2 tasks
1      storage/queue.go:381
1      storage/intent_resolver.go:275
ec2-54-209-150-36.compute-1.amazonaws.com:
0xc82031a380: 339 tasks
337    server/node.go:686
2      storage/queue.go:381
```

**Comment by tschottdorf** [comment]
we seem to be stuck (at or around the) command queue again:

```
2016/03/23 16:43:43.968269	1132.393221	node 2
16:43:43.968272	 .     3	... node 2
16:43:43.968308	 .    36	... executing 1 requests
16:43:43.968314	 .     6	... read-write path
16:43:43.968316	 .     2	... command queue
2016/03/23 16:43:40.725989	1135.635639	node 2
16:43:40.725992	 .     3	... node 2
16:43:40.726028	 .    36	... executing 1 requests
16:43:40.726034	 .     6	... read-write path
16:43:40.726035	 .     2	... command queue
2016/03/23 16:43:40.706464	1135.655300	node 2
16:43:40.706469	 .     5	... node 2
16:43:40.706540	 .    71	... executing 1 requests
16:43:40.706550	 .    10	... read-write path
16:43:40.706552	 .     2	... command queue
```

**Comment by tschottdorf** [comment]
[goroutine.ec2-54-209-69-52.txt](https://github.com/cockroachdb/cockroach/files/186473/goroutine.ec2-54-209-69-52.txt)

**Comment by tschottdorf** [comment]
One observation which may or may not be related to this problem (note that I'm not looking at the AdminSplit issue at all at the moment) is that `processIntentsAsync` pushes back onto the caller, but one of the callers is in a Raft command (`applyRaftCommand`). It would seem that there's potential for deadlock: we're waiting for something else to make it through Raft before we make it out of Raft. If the two involved key ranges overlap, we could be seeing something like this.

I don't think this is happening here because there's only one goroutine with `RunLimitedAsyncTask` in its stacktrace, and that is a successfully started task.

**Comment by tschottdorf** [comment]
It is a bit odd though that the longest-stalled goroutine should be the one related to intents:

```
1: semacquire [21 minutes] [Created by stop.(*Stopper).RunLimitedAsyncTask @ stopper.go:219]
    sync      sema.go:47              runtime_Semacquire(#3071)
    sync      waitgroup.go:127        (*WaitGroup).Wait(#3070)
    storage   replica.go:782          (*Replica).beginCmds(#114, #4059, #19)
    storage   replica.go:947          (*Replica).addWriteCmd(#114, #4780, #666, #4799, 0, #36, 0x2, 0x2a, 0, 0, ...)
    storage   replica.go:708          (*Replica).Send(#114, #4780, #666, #4799, 0, #36, 0x2, 0x2a, 0, 0, ...)
    storage   store.go:1617           (*Store).Send(#178, #4780, #666, #4799, 0, #36, 0x2, 0x2a, 0, 0, ...)
    storage   stores.go:174           (*Stores).Send(#245, #4780, #665, 0, 0, #36, 0x2, 0x2a, 0, 0, ...)
    server    node.go:674             (*Node).Batch.func2()
    stop      stopper.go:166          (*Stopper).RunTask(#220, #3321, 0)
    server    node.go:686             (*Node).Batch(#179, #4786, #906, #592, #905, 0, 0)
    kv        send.go:260             sendOne(#976, 0x16, #885, #4794, #100, 0, 0, #36, 0x2, 0x2a, ...)
    kv        send.go:154             send(#4780, #664, 0, #37, #35, #3692, 0x3, 0x3, 0, 0, ...)
    kv        dist_sender.go:340      (*DistSender).sendRPC(#297, #4780, #664, 0x2a, #3692, 0x3, 0x3, 0, 0, 0, ...)
    kv        dist_sender.go:429      (*DistSender).sendSingleRange(#297, #4780, #664, 0, 0, 0, 0, 0, 0, 0, ...)
    kv        dist_sender.go:667      (*DistSender).sendChunk.func1(#3323, #3322, #3325, #297, #3324, 0x1, 0x8)
    kv        dist_sender.go:668      (*DistSender).sendChunk(#297, #4780, #664, 0, 0, 0, 0, 0, 0, 0, ...)
    kv        dist_sender.go:525      (*DistSender).Send(#297, #4780, #664, 0, 0, 0, 0, 0, 0, 0, ...)
    kv        txn_coord_sender.go:415 (*TxnCoordSender).Send(#301, #4780, #664, 0, 0, 0, 0, 0, 0, 0, ...)
    client    db.go:419               (*DB).send(#312, 0, #1821, 0x1, 0x1, 0, 0)
    client    db.go:378               send)-fm(0, #1821, 0x1, 0x1, 0x1, 0)
    client    db.go:343               sendAndFill(#3326, #2631, #4389, #1820)
    client    db.go:378               (*DB).RunWithResponse(#312, #2631, 0x1, 0x1)
    storage   intent_resolver.go:216  (*intentResolver).maybePushTransactions(#292, #4786, #904, #2972, 0x1, 0x1, #4798, 0, 0, 0, ...)
    storage   intent_resolver.go:265  (*intentResolver).processIntentsAsync.func1()
    stop      stopper.go:218          (*Stopper).RunLimitedAsyncTask.func1(#220, #23, 0x1a, 0x113, #300, #1915)
```

**Comment by tschottdorf** [comment]
goroutine list, after filtering out everything that's stuck on the waitgroup or is a worker (`grep -E ^[0-9] pp.goroutine | grep -vE 'RunWorker|semacquire'`)

Looks pretty normal to me. @nvanbenschoten we're gonna need something that can help us in this situation (this is #5368 again).

```
1: running [Created by http2.(*serverConn).processHeaders @ server.go:1442]
1: IO wait [Created by grpc.(*Server).Serve @ server.go:261]
2: IO wait [3 minutes] [Created by grpc.(*Server).Serve @ server.go:261]
1: IO wait [Created by grpc.(*Server).Serve @ server.go:261]
1: IO wait [Created by grpc.(*Server).Serve @ server.go:261]
1: IO wait [10 minutes] [Created by grpc.(*Server).Serve @ server.go:261]
1: IO wait [Created by grpc.(*Server).Serve @ server.go:261]
1: IO wait [Created by transport.newHTTP2Client @ http2_client.go:200]
1: IO wait [Created by transport.newHTTP2Client @ http2_client.go:200]
3: IO wait [2 minutes] [Created by transport.newHTTP2Client @ http2_client.go:200]
1: IO wait [Created by transport.newHTTP2Client @ http2_client.go:200]
1: IO wait [Created by transport.newHTTP2Client @ http2_client.go:200]
1: IO wait [Created by http2.(*serverConn).serve @ server.go:700]
1: select [30 minutes] [Created by grpc.(*Server).serveStreams.func1 @ server.go:324]
1: select [106 minutes]
1: select [106 minutes] [locked]
3: select [106 minutes] [Created by grpc.(*Server).serveStreams.func1 @ server.go:324]
1: select [Created by http.(*Server).Serve @ server.go:2137]
7: select [106 minutes] [Created by grpc.NewConn @ clientconn.go:323]
1: syscall [106 minutes] [Created by signal.init.1 @ signal_unix.go:28]
1: chan receive [106 minutes] [Created by engine.(*RocksDB).Open @ rocksdb.go:136]
1: syscall [106 minutes] [locked]
1: chan receive [106 minutes] [Created by server.RegisterAdminHandlerFromEndpoint.func1 @ admin.pb.gw.go:181]
1: chan receive [Created by log.init.1 @ clog.go:604]
7: select [25 minutes] [Created by transport.newHTTP2Client @ http2_client.go:194]
7: select [15 minutes] [Created by transport.newHTTP2Server @ http2_server.go:134]
100: select [86 minutes] [Created by grpc.NewClientStream @ stream.go:156]
```

**Comment by tschottdorf** [comment]
back to the split issue:

The Split queue tries to split this range:

```
[split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13")
```

and `AdminSplit` computes the split key to be the `StartKey` `/System/StatusNode/4`, which is clearly not a valid choice. The process repeats indefinitely.

**Comment by tschottdorf** [comment]
There's what I think amounts to a bug in `MVCCFindSplitKey` where it does this comparison

```
	if bestSplitKey.Equal(encStartKey) {
		return nil, util.Errorf("the range cannot be split; considered range %q-%q has no valid splits", key, endKey)
	}
```

the bug is that both are MVCCMetadataKeys, but `encStartKey` has a zero timestamp (because that's just how we create it) whereas the first key, which is equal to `encStartKey.Key`, has a nonzero timestamp, so the assertion wouldn't fire. That would explain why we're attempting the split in the first place, but not why there isn't a better candidate.

**Comment by tschottdorf** [comment]
https://github.com/cockroachdb/cockroach/pull/5516 showed that we've got a range so full of versions of that above key that the split key is one of them.
It also seems like (but maybe I'm missing something - I raced through this stuff) we don't make sure that we don't split in between MVCC versions of the same key; no bueno.

**Comment by mberhault** [comment]
ok, I'm now getting:
```
I160324 13:38:25.980849 storage/split_queue.go:113  splitting range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13") size=67740735 max=67108864
E160324 13:38:26.077455 storage/queue.go:444  [split] range=54 [/System/StatusNode/4-/System/"tsd\x12cr.node.exec.latency-10m-p90\x00\x01\x89\xf8\x06.\xc13"): error: storage/replica_command.go:1573: unable to determine split key: storage/engine/mvcc.go:1821: the range cannot be split; considered range "/System/StatusNode/4"-"/System/\"tsd\\x12cr.node.exec.latency-10m-p90\\x00\\x01\\x89\\xf8\\x06.\\xc13\"" has no valid splits
```

Which seems like the right thing from your fix. Should we bail out more gracefully when we can't split anyway? Any chance we could try StartKey+1 as long as it's in the range?

**Comment by tamird** [comment]
Could we address this by setting a different GC policy for these keys?

**Comment by bdarnell** [comment]
Splitting at StartKey+1 might allow this split to proceed, but we don't really want it to if it's motivated by temporary GC'able data: we aren't currently able to merge the ranges back together once they are split down to single keys.

Setting a shorter GC period for `/System/StatusNode/` may do the trick, but we may also need to run the GC process more frequently. If a node has a lot of GC'able data it would be better to GC it than to split it, and it may be worth doing a single-key GC on a random sample of writes to the key. 

How long does it take for these errors to start showing up, and how high have you seen the size at the end of the `split_queue.go:113` log line get? This could be the cause of #5467.

**Comment by tschottdorf** [comment]
We discussed offline and it seems that at least for the store/node status keys, we can simply write an inline key (though that could backfire if we needed to run transactions on that, ever).

Grepped for your log line in node2.log.parse.txt from #5467: 

```
I160321 16:53:39.971986 storage/split_queue.go:113  splitting range=25 [/Table/53/1/"\x16ݓi\xba\x14O)\x9aP\x9bA\x9e$U\x8b"/"\xfalX\xe4u/E~\xaeς44ma\xb0"-/Table/53/1/"MP\xf5R\xe2\x13F<\x88;y\xe0\xd4Q\xc0\x1d"/"\xd1jP\x9d\x99\xccO[\x8e\x1c3A7bͻ") size=67108922 max=67108864
I160321 16:56:27.031759 storage/split_queue.go:113  splitting range=27 [/Table/53/1/"\x82P\xeb\x17%iK\xf3\x9b\xfa\xfd\x0e\xcd\x1a\x16}"/"ء\x9f\xccS%D՟\xf1\x03\xf7\xb8o\xa5\xf6"-/Table/53/1/"\xb7#\xe6'VeKU\xa0\x8c\x8d,\x82\x80&\xd4"/"=S\x87\x95\x80CM.\x87\xa4\xd5\xfe1mp\xd8") size=67109209 max=67108864
I160321 17:08:20.698970 storage/split_queue.go:113  splitting range=23 [/Table/53/1/""/"\xc5\t\xeb\x1f\x89mAD\xb7\xd1F\x0e\xb7Lp("-/Table/53/1/"\x16ݓi\xba\x14O)\x9aP\x9bA\x9e$U\x8b"/"\xfalX\xe4u/E~\xaeς44ma\xb0") size=67110105 max=67108864
I160321 17:42:48.560217 storage/split_queue.go:113  splitting range=21 [/Table/53-/Table/53/1/""/"bZjy<\x85N@\xa3o\x14\xa9\xb1\xf0J\xf3") size=67109442 max=67108864
I160321 20:35:26.985727 storage/split_queue.go:113  splitting range=32 [/Table/53/2/""/Mon Mar 21 15:17:08 UTC 2016-/Max) size=67108999 max=67108864
I160321 20:51:06.044221 storage/split_queue.go:113  splitting range=25 [/Table/53/1/"\x16ݓi\xba\x14O)\x9aP\x9bA\x9e$U\x8b"/"\xfalX\xe4u/E~\xaeς44ma\xb0"-/Table/53/1/"2\x1d\x82\x139\xd4@\x0f\xb18\x90\x06\x9c\xc1\x83w"/"\x03\xdfc\xd3B\tO\x8b\x8a%ȡ\x9f\xd7\xf4\xf5") size=67108869 max=67108864
I160321 21:26:31.364718 storage/split_queue.go:113  splitting range=62 [/Table/53/1/""/"\xfb\bx\x89\xac5F\xe6\x8e]-\xe5\x81V\xefe"-/Table/53/1/"\x16ݓi\xba\x14O)\x9aP\x9bA\x9e$U\x8b"/"\xfalX\xe4u/E~\xaeς44ma\xb0") size=67109898 max=67108864
I160321 21:54:31.284547 storage/split_queue.go:113  splitting range=11 [/Table/53/1/"\xebc\xf7}\xefdLų\xee\xe1\x17\xcb\x1c!\x99"/"|\xb3\xa2\x11\x82\x9aL\x90\xa8\xd8*87!\x13b"-/Table/53/2/""/Mon Mar 21 15:17:08 UTC 2016) size=67109154 max=67108864
I160322 00:47:35.946642 storage/split_queue.go:113  splitting range=23 [/Table/53/1/""/"\xc5\t\xeb\x1f\x89mAD\xb7\xd1F\x0e\xb7Lp("-/Table/53/1/""/"\xfb\bx\x89\xac5F\xe6\x8e]-\xe5\x81V\xefe") size=67109488 max=67108864

// 3hrs pass without such a message

I160322 03:54:07.040343 /go/src/google.golang.org/grpc/clientconn.go:318  Failed to dial ip-172-31-49-146:26257: grpc: timed out trying to connect; please retry.
fatal error: runtime: cannot allocate memory
```

```
As for #5501 (this one):

16:25 $ grep -oE 'size=[0-9]+' ~/5501/logs.*/*.stderr | sed 's/.*size\=//' | sort -r | head -n 10
68492687
68492687
68492255
68484636
68484204
68484204
68483772
68483340
68483340
68483340
```

I would take that to mean that it probably doesn't have anything to do with it.

**Comment by bdarnell** [comment]
The #5467 crash was while applying a snapshot, which means that it involves a range that was not present on node 2 prior to the crash, so the absence of split queue messages on the node doesn't eliminate the possibility that these are the same bug. However, the "already split" message isn't showing up in any of the node logs from that bug so it's apparently not related.

**Comment by tamird** [comment]
I'm going to close this for now, the proximate cause of problems was fixed in #5595.
